packages/expo-haptics/android/src/main/java/expo/modules/haptics/HapticsModule.kt

```diff
@@ -4,42 +4,45 @@ import android.content.Context
 import android.os.Build
 import android.os.VibrationEffect
 import android.os.Vibrator
-import expo.modules.core.ExportedModule
-import expo.modules.core.Promise
-import expo.modules.core.interfaces.ExpoMethod
-import expo.modules.haptics.arguments.HapticsInvalidArgumentException
 import expo.modules.haptics.arguments.HapticsImpactType
+import expo.modules.haptics.arguments.HapticsInvalidArgumentException
 import expo.modules.haptics.arguments.HapticsNotificationType
 import expo.modules.haptics.arguments.HapticsSelectionType
 import expo.modules.haptics.arguments.HapticsVibrationType
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.exception.Exceptions
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
 
-class HapticsModule internal constructor(context: Context) : ExportedModule(context) {
-  private val mVibrator = context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
-  override fun getName() = "ExpoHaptics"
+class HapticsModule : Module() {
+  private val context: Context
+    get() = appContext.reactContext ?: throw Exceptions.ReactContextLost()
+  private val mVibrator get() = context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
 
-  @ExpoMethod
-  fun notificationAsync(type: String, promise: Promise) {
-    try {
-      vibrate(HapticsNotificationType.fromString(type))
-      promise.resolve(null)
-    } catch (e: HapticsInvalidArgumentException) {
-      promise.reject(e)
-    }
-  }
+  override fun definition() = ModuleDefinition {
+    Name("ExpoHaptics")
 
-  @ExpoMethod
-  fun selectionAsync(promise: Promise) {
-    vibrate(HapticsSelectionType)
-    promise.resolve(null)
-  }
+    AsyncFunction("notificationAsync") { type: String, promise: Promise ->
+      try {
+        vibrate(HapticsNotificationType.fromString(type))
+        promise.resolve(null)
+      } catch (e: HapticsInvalidArgumentException) {
+        throw HapticsInvalidArgumentException(e.message)
+      }
+    }
```

Comment:
- We don't need the promise here 😁 it will be added under the hood by Expo Modules
- Also, rethrowing the same error doesn't make sense, so we can get rid of the `try-catch`
```suggestion
    AsyncFunction("notificationAsync") { type: String ->
      vibrate(HapticsNotificationType.fromString(type))
    }
```